zebra/zebra_nhg.c

```diff
@@ -2210,24 +2266,22 @@ static int nexthop_active(struct nexthop *nexthop, struct nhg_hash_entry *nhe,
 		}
 
 		if (match->type == ZEBRA_ROUTE_CONNECT) {
-			/* Directly point connected route. */
-			newhop = match->nhe->nhg.nexthop;
-			if (newhop) {
-				if (nexthop->type == NEXTHOP_TYPE_IPV4
-				    || nexthop->type == NEXTHOP_TYPE_IPV6)
-					nexthop->ifindex = newhop->ifindex;
-				else if (nexthop->ifindex != newhop->ifindex) {
-					if (IS_ZEBRA_DEBUG_RIB_DETAILED)
-						zlog_debug(
-							"%s: %pNHv given ifindex does not match nexthops ifindex found found: %pNHv",
-							__func__, nexthop,
-							newhop);
-					/*
-					 * NEXTHOP_TYPE_*_IFINDEX but ifindex
-					 * doesn't match what we found.
-					 */
-					return 0;
-				}
+			newhop = zebra_nhg_connected_ifindex(rn, match,
+							     nexthop->ifindex);
+
+			if (nexthop->type == NEXTHOP_TYPE_IPV4 ||
+			    nexthop->type == NEXTHOP_TYPE_IPV6)
+				nexthop->ifindex = newhop->ifindex;
+			else if (nexthop->ifindex != newhop->ifindex) {
+				if (IS_ZEBRA_DEBUG_RIB_DETAILED)
+					zlog_debug(
+						"%s: %pNHv given ifindex does not match nexthops ifindex found found: %pNHv",
```

Comment:
heh - maybe could remove one of the "found found" words?

Reply:
fixed